.github/workflows/promote_to_draft.yml

```diff
@@ -0,0 +1,100 @@
+name: 🛫  Promote a Dataset to Draft
+run-name: "🛫 Promote a Dataset to Draft: ${{ inputs.product }}"
+
+on:
+  workflow_dispatch:
+    inputs:
+      product:
+        description: "Data product name (folder name in S3 publishing bucket)"
+        type: choice
+        required: true
+        options:
+          - db-cbbr
+          - db-checkbook
+          - db-colp
+          - db-cpdb
+          - db-developments
+          - db-facilities
+          - db-pluto
+          - db-zoningtaxlots
+          - db-template
+      build:
+        description: "Build folder to promote"
+        type: string
+        required: true
+      draft_summary:
+        description: "Draft revision summary (becomes a part of draft name in s3)"
+        type: string
+        required: false
+        default: ''
+      product_acl:
+        description: "Set access control list for output in DO"
+        type: choice
+        required: true
+        default: public-read
+        options:
+          - public-read 
+          - private
+
+jobs:
+  promote:
+    runs-on: ubuntu-22.04
+    defaults:
+      run:
+        shell: bash
+    container:
+      image: nycplanning/build-base:latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            dcpy 
+            bash
+            python
+
+      - name: Load Secrets
+        uses: 1password/load-secrets-action@v1
+        with:
+          export-env: true
+        env:
+          OP_SERVICE_ACCOUNT_TOKEN: ${{ secrets.OP_SERVICE_ACCOUNT_TOKEN }}
+          AWS_S3_ENDPOINT: "op://Data Engineering/DO_keys/AWS_S3_ENDPOINT"
+          AWS_SECRET_ACCESS_KEY: "op://Data Engineering/DO_keys/AWS_SECRET_ACCESS_KEY"
+          AWS_ACCESS_KEY_ID: "op://Data Engineering/DO_keys/AWS_ACCESS_KEY_ID"
+
+      - name: Finish container setup
+        working-directory: ./
+        run: ./bash/docker_container_setup.sh
+
+      - name: Promote Build to Draft
+        run: |
+          python3 -m dcpy.connectors.edm.publishing promote_to_draft \
+            --product ${{ inputs.product }} \
+            --build ${{ inputs.build }} \
+            --draft-summary ${{ inputs.draft_summary }} \
+            --acl ${{ inputs.product_acl }}
+
+      - name: Download Metadata
+        run: |
+          python3 -m dcpy.connectors.edm.publishing download_file \
+            --filepath build_metadata.json \
+            --product ${{ inputs.product }} \
+            --product-type build \
+            --version ${{inputs.build}}
+
+      - name: Set version and build sha
+        run: |
+          echo "version=$(cat build_metadata.json | jq -r .version)" >> $GITHUB_ENV
+          echo "commit=$(cat build_metadata.json | jq -r .commit)" >> $GITHUB_ENV
+
+      - name: create tag
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GHP_TOKEN }}
```

Comment:
Maybe you ran into issue because perhaps a standard action token can _create_ a tag but not _overwrite_ one? If the tag you were creating existed?

That being said, I'm not 100% sure we need to tag when promoting to draft. I guess maybe there's not much downside, but I think maybe keeping tags specifically for datasets that we've published might make more sense